cli-rs/src/main.rs

```diff
@@ -260,63 +272,40 @@ fn run_command(command: Commands) {
 }
 
 #[allow(clippy::too_many_arguments)]
-fn compile_rust<F: FieldElement>(
+fn compile_rust(
     file_name: &str,
+    field: KnownField,
     output_dir: &Path,
     coprocessors: Option<String>,
     continuations: bool,
 ) -> Result<(), Vec<String>> {
-    let mut runtime = match coprocessors {
-        Some(list) => {
-            powdr_riscv::Runtime::try_from(list.split(',').collect::<Vec<_>>().as_ref()).unwrap()
-        }
-        None => powdr_riscv::Runtime::base(),
-    };
-
-    if continuations {
-        if runtime.has_submachine("poseidon_gl") {
-            return Err(vec![
-                "Poseidon continuations mode is chosen automatically and incompatible with the chosen standard Poseidon coprocessor".to_string(),
-            ]);
-        }
-        runtime = runtime.with_poseidon_for_continuations();
-    }
-
-    powdr_riscv::compile_rust::<F>(file_name, output_dir, true, &runtime, continuations, None)
+    let libs = coprocessors_to_options(coprocessors)?;
+    let options = CompilerOptions::new(field, libs, continuations);
+    powdr_riscv::compile_rust(file_name, options, output_dir, true, None)
         .ok_or_else(|| vec!["could not compile rust".to_string()])?;
 
     Ok(())
 }
 
-fn compile_riscv_elf<F: FieldElement>(
+fn compile_riscv_elf(
     input_file: &str,
+    field: KnownField,
     output_dir: &Path,
     coprocessors: Option<String>,
     continuations: bool,
 ) -> Result<(), Vec<String>> {
-    let runtime = match coprocessors {
-        Some(list) => {
-            powdr_riscv::Runtime::try_from(list.split(',').collect::<Vec<_>>().as_ref()).unwrap()
-        }
-        None => powdr_riscv::Runtime::base(),
-    };
-
-    powdr_riscv::compile_riscv_elf::<F>(
-        input_file,
-        Path::new(input_file),
-        output_dir,
-        true,
-        &runtime,
-        continuations,
-    )
-    .ok_or_else(|| vec!["could not translate RISC-V executable".to_string()])?;
+    let libs = coprocessors_to_options(coprocessors)?;
+    let options = CompilerOptions::new(field, libs, continuations);
+    powdr_riscv::compile_riscv_elf(input_file, Path::new(input_file), options, output_dir, true)
+        .ok_or_else(|| vec!["could not translate RISC-V executable".to_string()])?;
 
     Ok(())
 }
 
 #[allow(clippy::too_many_arguments)]
 fn execute<F: FieldElement>(
```

Comment:
execute already takes `F`, it doesn't need the `field: KnownField` parameter

Reply:
yea I kept both because they are used in different ways, so didn't want to mix them up. Fine with removing, but want to double check after explaining why. Should I still remove it?

Reply:
yes, i'd call `F::known_field` inside

Reply:
there's a similar situation in another part of the code, but there I think `F` can be removed, because there you `match` on the known field to then pick a concrete field

Reply:
done here. Can you comment on the other instance?

Reply:
actually this one ended up being removed completely when I changed the other one